Mod/sergioarmgpl (#364)
cncf/mentoring: students/README.md

# CNCF Students toolkit
Is a group of resources, meeting ideas, and groups to get students involved in the CNCF ecosystem. They will be able to gain experience using cloud native technologies for their future jobs and develop skills to contribute in open source projects. It also creates a space for students and universities in the CNCF ecosystem.

>  ## How it works

Reviewer: Broken indent.

There are many different options that a student can choose to work with or learn about CNCF projects, the following sections describe a few of the different options.
 
### Working on issues
College students (primarily engineering students) have the task of working on issues of their interest.
 
### Working on thesis
A college student can work on a research topic for their own thesis that involves the implementation, or a test drive of cloud native technologies that solve real life problems, or shows different use cases for cloud native technologies.
 
### Class projects
A Professor can design a project that solves a real work problem and show how to uses cloud native technologies, the professor can explore ideas such as:
- Test new features for an specific CNCF project
- Implement a solution that solves a real life problem
- Develop new features for a CNCF project
 At the end, the best projects of the class can be presented to CNCF projects or into CNCF Community Groups. In addition, these projects develop skills that student need to apply to the Google Summer of Code.
 

> ### Speakers and other contributions

Reviewer: CNCF Mentoring programs (mentoring.cncf.io) is a better fit rather than the Ambassadors program.

These initiatives should not be only focused on coding. Students can also be speakers or social media contributors, like the [CNCF Ambassador Program](https://github.com/cncf/ambassadors).
 
### Open Syllabus
Professors can work on open content of their courses and start to include CNCF technologies in their classes and projects, generating content as workshops, tutorials, etc., created by professors or students.
 
### CNCF Students Day

> Professors and Universities can organize events to introduce students into CNCF technologies, including basic content as how to use containers, Kubernetes and complementary cloud native technologies. They can also attend [Cloud Native Community Groups](https://community.cncf.io/) or [Kubernetes Community Days](https://kubernetescommunitydays.org/) in their region. Many of these events have free or discounted tickets.

Reviewer 2: Some suggestion to make sentence easier to read.

- *to CNCF technologies*
- *content such as containers*

- [Realtime COVID-19 Display System Using Linkerd](https://www.youtube.com/watch?v=XWlpS78wRks)
 
### Open Syllabus and project
This repository is an initiative to develop an open content and resources for students to learn about cloud native technologies.
- [Operating System Syllabus and Projects](https://github.com/sergioarmgpl/operating-systems-usac-course)
 
### CNCF Students Day
This event was hosted in Nicaragua, as an online event, to promote CNCF technologies. The students from Guatemala and some other speakers that are organizers of CNCF Community Groups participated giving workshops, and talks focused on containers, Kubernetes, and cloud native in general. Most of the attendees were students.

- [Cloud Native Students Day URACCAN](https://community.cncf.io/events/details/cncf-cloud-nativegt-presents-cloud-native-students-day-uraccan/)
 

> ## How to promote this toolkit

Reviewer: Let's remove this section for now.

This toolkit or students movement should be promoted using the hashtag:
- #CNCFStudents  
  
Please share all students events with this hashtag to grow as a CNCF Students movement
 
## Contact
You can contact us in the CNCF #students slack channel for more information.
- [CNCF Slack](https://slack.cncf.io/)

>  ### Team

Reviewer: Broken indent.

> - Sergio Méndez, USAC

Reviewer: I think someone from CNCF staff should be responsible here as well. @xmulligan, you?

> # CNCF Students toolkit
> Is a group of resources, meeting ideas and groups to get involved seniors students into the CNCF ecosystem, in the way that they can get experience using cloud native technologies for their future jobs and develop skills to contribute in open source projects. It also creates a space for students and universities in the CNCF ecosystem.
>  ## How it works
> There are different options that a student can choose to work or learn about CNCF projects, the following sections describe the different options that can be suggested to work on.

Reviewer 3: ```suggestion
# CNCF Students toolkit
Is a group of resources, meeting ideas, and groups to get students involved in the CNCF ecosystem. They will be able to gain experience using cloud native technologies for their future jobs and develop skills to contribute in open source projects. It also creates a space for students and universities in the CNCF ecosystem.
 ## How it works
There are many different options that a student can choose to work with or learn about CNCF projects, the following sections describe a few of the different options.
```

> ### Working on issues
> The College students(primarily engineering students) have the task of working on issues of their interest.
>  
> ### Working on thesis
> A collage student can work on a research topic for its own thesis that involves the implementation, or a test drive of cloud native technologies that solve real life problems, or shows different use cases for cloud native technologies.

Reviewer 3: ```suggestion
### Working on issues
College students (primarily engineering students) have the task of working on issues of their interest.
 
### Working on thesis
A college student can work on a research topic for their own thesis that involves the implementation, or a test drive of cloud native technologies that solve real life problems, or shows different use cases for cloud native technologies.
```

> ### Class projects
> A Professor can design a project that resolves a real work problem and show how to uses cloud native technologies, the professor can explore ideas as:
> - Test new features for an specific CNCF project
> - Implement a solution that solves a real life problem
> - Develop new features for a CNCF project
>  At the end the best projects of the class can be presented to CNCF projects or into CNCF Community Groups. In addition these projects develop skills in order that a student can apply to the Google Summer of Code.

Reviewer 3: ```suggestion
### Class projects
A Professor can design a project that solves a real work problem and show how to uses cloud native technologies, the professor can explore ideas such as:
- Test new features for an specific CNCF project
- Implement a solution that solves a real life problem
- Develop new features for a CNCF project
 At the end, the best projects of the class can be presented to CNCF projects or into CNCF Community Groups. In addition, these projects develop skills that student need to apply to the Google Summer of Code.
```

> ### Speakers and other contributions
> This initiative should not be only focused on coding. Could be speakers or social media contributors in the same way like the CNCF Ambassador Program https://github.com/cncf/ambassadors.
>  
> ### Open Syllabus
> Professors can work on open content of their courses and start to include CNCF technologies in their classes and projects, generating content as workshops, tutorials, etc., created by professors or students.

Reviewer 3: ```suggestion
### Speakers and other contributions
These initiatives should not be only focused on coding. Students can also be speakers or social media contributors, like the [CNCF Ambassador Program](https://github.com/cncf/ambassadors).
 
### Open Syllabus
Professors can work on open content of their courses and start to include CNCF technologies in their classes and projects, generating content as workshops, tutorials, etc., created by professors or students.
```

> ### CNCF Students Day
> Professors and Universities can organize events to introduce students into CNCF technologies, including basic content as how to use containers, Kubernetes and complementary cloud native technologies.
>  
> # Involved roles
> This section describes the different roles that can help to get involved students into the CNCF Ecosystem.
>  
> ## Professors
> A professor is one of the primary roles to promote the students into the CNCF Ecosystem, leading students across projects, workshops, community groups and other activities to get involved the students into the CNCF Ecosystem.

Reviewer 3: ```suggestion
### CNCF Students Day
Professors and Universities can organize events to introduce students into CNCF technologies, including basic content as how to use containers, Kubernetes and complementary cloud native technologies. They can also attend [Cloud Native Community Groups](https://community.cncf.io/) or [Kubernetes Community Days](https://kubernetescommunitydays.org/) in their region. Many of these events have free or discounted tickets.
 
# Involved roles
This section describes the different roles that can help to get involved students into the CNCF Ecosystem.
 
## Professors
A professor is one of the primary roles to promote the students into the CNCF Ecosystem, leading students across projects, workshops, community groups and other activities to get involved the students into the CNCF Ecosystem.
```

> # Involved roles
> This section describes the different roles that can help to get involved students into the CNCF Ecosystem.
>  
> ## Professors
> A professor is one of the primary roles to promote the students into the CNCF Ecosystem, leading students across projects, workshops, community groups and other activities to get involved the students into the CNCF Ecosystem.

Reviewer 3: ```suggestion
# Involved roles
This section describes the different roles that can help get students involved into the CNCF Ecosystem.
 
## Professors
A professor is one of the primary roles to promote the students into the CNCF Ecosystem, leading students across projects, workshops, community groups, and other activities to get involved the students into the CNCF Ecosystem.
```

> ## Mentor
> A mentor that can be an ambassador or some active contributor of a CNCF of election. This CNCF Mentor here plays a vital role in helping the students work on topics of their interest related to these tasks, developing in that way their skills for the industry.  These mentors help universities or students to lead the research.
>  ## University
> Universities can promote these activities with their professors and students, doing the previous described activities, with the help of mentors to lead these activities in the best way possible.

Reviewer 3: ```suggestion
## Mentor
A mentor that can be an ambassador or another active contributor to a CNCF project. This CNCF Mentor here plays a vital role in helping the students work on topics of their interest related to these projects and developing their skills for the industry.  These mentors help universities or students lead their research.
## University
Universities can promote these activities with their professors and students, doing the previous described activities, with the help of mentors to lead these activities in the best way possible.
```

> ## Community
> CNCF Community Groups can be a safe place for students to present how they are using technologies, present projects, do workshops, etc.
>  ## Benefits of this toolkit
> - Helps CNCF in attracting college students to contribute to their projects.
> - Helps CNCF in attracting college students to use cloud native technologies.
> - Helps CNCF projects to test new features to speed their release and testing times.
> - Involve Students in programs like Community Bridge or Google Summer of Code to contribute with CNCF projects.
> - Open a space for students to get jobs related to cloud native technologies.
> - Create a space for Universities' contribution by doing research with their students using CNCF projects.
> - Create a space for innovation and new ideas where universities, students and CNCF can get involved.

Reviewer 3: ```suggestion
## Community
CNCF Community Groups can be a safe place for students to present how they are using technologies, present projects, do workshops, and learn from others in the community.
## Benefits of this toolkit
- Helps CNCF in attracting college students to contribute to their projects.
- Helps CNCF in attracting college students to use cloud native technologies.
- Helps CNCF projects to test new features to speed their release and testing times.
- Involve Students in programs like Community Bridge or Google Summer of Code to contribute to CNCF projects.
- Open a space for students to get jobs related to cloud native technologies.
- Create a space for Universities' contribution by doing research with their students using CNCF projects.
- Create a space for innovation and new ideas where universities, students, and CNCF  projects can get involved.
```

> - Promote diversity and inclusion in the way that universities and students in other countries can contribute.
>  ## To be done
> - List CNCF projects that could be interesting in this student toolkit
> - Define the minimum skills to develop in potential students contributors
> - Define the steps to get involved students into participate into GSoC related to CNCF projects
> - List professors, universities that can be interested in this program.
> - Generate a vault of projects, features of issues that the students/students can be involved in
> - If possible create a CNCF conference for students to promote students participation
> - Gather students opinions about this kind of opportunities to enrich this student kit
> - Work together with universities to update their syllabus in the way to include cloud native theory in their courses.
> - Create Bi-Weekly Work Meetings to develop and promote this toolkit

Reviewer 3: ```suggestion
- Promote diversity and inclusion in the way that universities and students in other countries can contribute.
 ## To be done
- List CNCF projects that could be interesting in this student toolkit
- Define the minimum skills to develop in potential students contributors
- Define the steps to get involved students into participate into GSoC related to CNCF projects
- List professors, universities that can be interested in this program.
- Generate a vault of projects, features of issues that the students/students can be involved in
- If possible create a CNCF conference for students to promote students participation
- Gather students opinions about this kind of opportunities to enrich this student kit
- Work together with universities to update their syllabus in the way to include cloud native theory in their courses.
- Create Bi-Weekly Work Meetings to develop and promote this toolkit
```

> ## Students Event Examples
> ### Students Presenting a Class Project
> The students of USAC, university of Guatemala presented a project about a high concurrency system that displays real time information about Covid-19, this project used Cloud Native technologies such as Kubernetes, Linkerd, gRPC, Prometheus and others. Charles Pretzer from Linkerd Community and Field Engineer of Buoyant, helped us answering questions and promoted students to Join the Linkerd Community. This was one of many sessions that try to involve students into the CNCF ecosystems, we also promoted the hashtag #cncfstudents to share students work on the social networks, my role was to organize the event and help the students to present the project to the linkerd people.
> - [Realtime COVID-19 Display System Using Linkerd](https://www.youtube.com/watch?v=XWlpS78wRks)

Reviewer 3: ```suggestion
## Students Event Examples
### Students Presenting a Class Project
The students of USAC, University of Guatemala presented a project about a high concurrency system that displays real time information about Covid-19, this project used Cloud Native technologies such as Kubernetes, Linkerd, gRPC, Prometheus, and others. Charles Pretzer from Linkerd Community and Field Engineer of Buoyant, helped us answering questions and promoted students to Join the Linkerd Community. This was one of many sessions that try to involve students into the CNCF ecosystems, we also promoted the hashtag #cncfstudents to share students work on the social networks, Sergio's role was to organize the event and help the students to present the project to the linkerd people.
- [Realtime COVID-19 Display System Using Linkerd](https://www.youtube.com/watch?v=XWlpS78wRks)
```

> ### Open Syllabus and project
> This repository is an initiative to develop an open content and resources for students to learn cloud native technologies.
> - [Operating System Syllabus and Projects](https://github.com/sergioarmgpl/operating-systems-usac-course)
>  
> ### CNCF Students Day
> This event was hosted in Nicaragua, as an online event, to promote CNCF technologies. The students from Guatemala and some other speakers that are organizers of CNCF Community Groups participated giving workshops, and talks focused on containers, kubernetes and cloud native in general, and most of the attendees were students.
> 
> - [Cloud Native Students Day URACCAN
> ](https://community.cncf.io/events/details/cncf-cloud-nativegt-presents-cloud-native-students-day-uraccan/)

Reviewer 3: ```suggestion
### Open Syllabus and project
This repository is an initiative to develop an open content and resources for students to learn about cloud native technologies.
- [Operating System Syllabus and Projects](https://github.com/sergioarmgpl/operating-systems-usac-course)
 
### CNCF Students Day
This event was hosted in Nicaragua, as an online event, to promote CNCF technologies. The students from Guatemala and some other speakers that are organizers of CNCF Community Groups participated giving workshops, and talks focused on containers, Kubernetes, and cloud native in general. Most of the attendees were students.

- [Cloud Native Students Day URACCAN](https://community.cncf.io/events/details/cncf-cloud-nativegt-presents-cloud-native-students-day-uraccan/)
```

 
## How to promote this toolkit
This toolkit or students movement should be promoted using the hashtag:
- #CNCFStudents  
  
Please share all students events with this hashtag to grow as a CNCF Students movement
 

> ## Contact
> You can contact us in the CNCF #students slack channel for more information.
> - [CNCF Slack](https://slack.cncf.io/)
>  ### Team
> - Sergio Méndez, USAC
> - Santiago Torres, in-toto Lead & TUF Core Developer

Reviewer 3: ```suggestion
## Contact
You can contact us in the CNCF #students slack channel for more information.
- [CNCF Slack](https://slack.cncf.io/)
### Team
- Sergio Méndez, USAC
- Santiago Torres, in-toto Lead & TUF Core Developer
```
